benchexec/tools/coccinelle.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import subprocess
+
+import benchexec.result as result
+from benchexec.tools.template import ToolNotFoundException, BaseTool2
+
+
+class Tool(BaseTool2):
+    """
+    Tool info for Coccinelle
+    (https://github.com/coccinelle/coccinelle).
+    """
+
+    REQUIRED_PATHS = ["standard.h" "ocaml" "standard.iso" "ocaml"]
+
+    def name(self):
+        return "Coccinelle"
+
+    def executable(self, tool_locator):
+        exe = tool_locator.find_executable("spatch")
+        dir_name = os.path.dirname(exe)
+        logging.debug("Looking in %s for spatch", dir_name)
+        for _, dir_names, file_names in os.walk(dir_name):
+            if "spatch" in file_names:
+                return exe
+            break
+        msg = (
+            f"ERROR: Did find a spatch in {os.path.dirname(exe)} "
+            f"but no 'spatch' directory besides it"
+        )
```

Comment:
Can you please explain a little bit about what this is supposed to do and why?

How does the existence of any file with name `spatch` somewhere deep inside the directory hierarchy next to the executable have an effect on running Coccinelle?

Why does the code look for files, but the error message is about a directory?

Furthermore, I think this check is completely redundant, because it will never fail, because there is always a file named `spatch` - the executable file itself that is searched above.